### ISSUE TITLE
AttributeError `is_visible` on `Dataset` for reuse

### ISSUE THREAD

**Issue by ThibaudDauce** [opendatateam/udata#3023]
https://errors.data.gouv.fr/organizations/sentry/issues/138748

Reproduced on https://www.data.gouv.fr/fr/reuses/snails-production-and-trade-in-france/

First theory: this reuse is link to a deleted dataset, so there is two options:
- This is not possible (we should update the db to remove this dataset ref inside this reuse) and try to find why this ref was still there…
- This is possible and we should add a check to check if the dataset exists before trying to get `.is_visible`.

I think I prefer the first option, since it's less code. (and the second option could lead to a lot of checks everywhere if we do not trust that our db is sane)

**Comment by maudetes** [comment]
Indeed. It's also documented in https://github.com/datagouv/data.gouv.fr/issues/478#issuecomment-2063745029.

We should apply a migration to remove deleted datasets from reuses, and make sure it can't happen again.
I think it shouldn't occur since https://github.com/opendatateam/udata/commit/4a8f3cfcd5e707e67094a360dfbf246a6b75845c, that is more recent than this dataset.

I think we can close this ticket as duplicate and prioritize the other one if it's okay?

**Comment by ThibaudDauce** [comment]
Yes!
